drivers/crypto/crypto_cc23x0.c

```diff
@@ -102,17 +121,42 @@ static void crypto_cc23x0_isr(const struct device *dev)
 	AESClearInterrupt(status);
 }
 
-static void crypto_cc23x0_cleanup(const struct device *dev)
+#ifdef CONFIG_CRYPTO_CC23X0_DMA
```

Comment:
Minor suggestion: I would use a single shared guard between `crypto_cc23x0_dma_enable` and `crypto_cc23x0_cleanup_dma` in order to have less `#ifdef`/`#endif`